alacritty_terminal/src/grid/mod.rs

```diff
@@ -158,22 +158,18 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
         }
     }
 
-    pub fn buffer_to_visible(&self, point: impl Into<Point<usize>>) -> Point<usize> {
+    pub fn buffer_to_visible(&self, point: impl Into<Point<usize>>) -> Option<Point<usize>> {
         let mut point = point.into();
 
-        let offset = point.line.saturating_sub(self.display_offset);
```

Comment:
Is it safe to removes `saturating_sub` here?

Reply:
If my math is correct, it should be. The `saturating_sub` was necessary since we removed the display offset from the selected line. Even that should in theory be safe without `saturating_sub`, because the line uses buffer indexing, meaning a visible line should always be >= display_offset.

However, we now add the number of visible lines and the display offset together, meaning that it should be double safe.